subprojects/docs/src/docs/userguide/organizingGradleProjects.adoc

```diff
@@ -15,55 +15,122 @@
+
+* It defines the directory `src/main/java` as the default source directory for compilation.
+* The output directory for compiled source code and other artifacts (like the JAR file) is `build`.
 
-When you run Gradle, it checks for the existence of a directory called `buildSrc`. Gradle then automatically compiles and tests this code and puts it in the classpath of your build script. You don't need to provide any further instruction. This can be a good place to add your custom tasks and plugins.
+By sticking to the default conventions, new developers to the project immediately know how to find their way around. While those conventions can be reconfigured, it makes it harder to build script users and authors to manage the build logic and its outcome. Try to stick to the default conventions as much as possible except if you need to adapt to the layout of a legacy project. Refer to the reference page of the relevant plugin to learn about its default conventions.
 
-For multi-project builds there can be only one `buildSrc` directory, which has to be in the root project directory.
+=== Always define a setting file
 
-Listed below is the default build script that Gradle applies to the `buildSrc` project:
+Gradle tries to locate a `settings.gradle` (Groovy DSL) or a `settings.gradle.kts` (Kotlin DSL) file with every invocation of the build. For that purpose, the runtime walks the hierarchy of the directory tree up to the root directory. The algorithm stops searching as soon as it finds the settings file.
 
+Always add a `settings.gradle` to the root directory of your build to avoid the initial performance impact. This recommendation applies to single project builds as well as multi-project builds. The file can either be empty or define the desired name of the project.
+
+A typical Java-based project could look as such:
+
+----
+.
+├── build.gradle
+├── settings.gradle
+└── src
+    └── main
+        └── java
+            └── ...
+----
+
+[[sec:split_build_logic_functional_boundaries]]
+=== Split build logic based on functional boundaries
+
+Over time build scripts can grow large in size if you continue to add logic over time. A good middle ground between a `build.gradle` file and a binary plugin is to extract a <<sec:script_plugins,script plugin>>. A script plugin can come in handy if you want to split the build logic along functional boundaries. Typical examples include publishing code, IDE configuration or a dedicated script plugin for declaring dependencies. It is highly recommended to identify those portions of the build code and separate them into a script plugin.
+
+The following example demonstrates the practice of link:https://en.wikipedia.org/wiki/Separation_of_concerns[separation of concerns] for build logic.
```

Comment:
I think the words "separation of concerns" stand by themselves without an extra link.

Reply:
done